### PR TITLE
(IMAGES-985) Update PsWindowsUpdate and other fixes

### PR DESCRIPTION
**Pull request by jcoconnor**
Series of fixes in order to progress this (November) refresh of Windows platforms:
1. IMAGES-985 - Upgrade PsWindowsUpdate for PS3+
2. Fix the Windows 2008 Build to adopt recent Pester additions.
3. Don't install .Net 4.7 for Win-2008r2/Win-7 (PS2 only) and Win-10/2016.

### CODE REVIEW

File: templates/win/common/scripts/common/windows-env.ps1 [hunk 1]
```diff
@@ -676,14 +676,20 @@ Function Install-PSWindowsUpdate {
 
 # Helper to import PsWindowsUpdate Module.
 Function Import-PsWindowsUpdateModule {
-  Write-Output "Importing PSWindowsUpdate module"
+
+  # Powershell 2.0 requires the use of the "Import-Module" command, whereas later versions
+  # auto-import provided the PSModulePath env variable contains the module path.
```
Comment (glennsarti) [inline templates/win/common/scripts/common/windows-env.ps1]: Can probably just always import the module.  No performance hit and reduces branching.

File: templates/win/common/scripts/provisioners/initiate-windows-update.ps1 [hunk 2]
```diff
@@ -39,7 +45,7 @@ Write-Output "========== A log and update report will be given at the end of the
 
 # Need to pick up Admin Username/Password from Environment for sched task
 Write-Output "Create Bootstrap Scheduled Task"
-schtasks /create /tn PackerWinUpdate /rl HIGHEST /ru "$ENV:ADMIN_USERNAME" /RP "$ENV:ADMIN_PASSWORD" /IT /F /SC ONSTART /DELAY 0000:20 /TR 'cmd /c c:\WINDOWS\system32\WindowsPowerShell\v1.0\powershell.exe -sta -WindowStyle Normal -ExecutionPolicy Bypass -NonInteractive -NoProfile -File C:\Packer\Scripts\packer-windows-update.ps1 >> C:\Packer\Logs\windows-update.log'
+schtasks /create /tn PackerWinUpdate /rl HIGHEST /ru "$ENV:ADMIN_USERNAME" /RP "$ENV:ADMIN_PASSWORD" /IT /F /SC ONSTART /DELAY 0000:50 /TR 'cmd /c c:\WINDOWS\system32\WindowsPowerShell\v1.0\powershell.exe -sta -WindowStyle Normal -ExecutionPolicy Bypass -NonInteractive -NoProfile -File C:\Packer\Scripts\packer-windows-update.ps1 >> C:\Packer\Logs\windows-update.log'
```
Comment (glennsarti) [inline templates/win/common/scripts/provisioners/initiate-windows-update.ps1]: Probably want to mention why the delay is increased in the commit msg

Comment (jcoconnor) [reply templates/win/common/scripts/provisioners/initiate-windows-update.ps1]: Good point - will do

**Comment by jcoconnor** [comment]
This PR has been put through a full build of all the ```vmpoooler``` images over the weekend with the exception of ```win-2008-x86_64```.
The ```win-2008-x86_64``` is still running but has just completed the base and windows update install phase and has just passed the windows update verification tests:
```
12:30:24     win-2008-x86_64-vmware-vcenter: Executing all tests in 'C:\Packer\Acceptance\windows-update\'
12:30:24     win-2008-x86_64-vmware-vcenter:
12:30:24     win-2008-x86_64-vmware-vcenter: Executing script C:\Packer\Acceptance\windows-update\WindowsUpdate.Tests.ps1
12:30:47     win-2008-x86_64-vmware-vcenter: Reboot is not needed
12:30:47     win-2008-x86_64-vmware-vcenter:
12:30:47     win-2008-x86_64-vmware-vcenter:   Describing Windows Update Validation Tests
12:30:47     win-2008-x86_64-vmware-vcenter:     [+] Should Have at least One recent Windows Update installed 23.04s
12:30:47     win-2008-x86_64-vmware-vcenter:     [+] Should not have any Windows Updates Pending 21ms
12:30:47     win-2008-x86_64-vmware-vcenter:     [+] Reboot Status should not be pending 14ms
12:30:47     win-2008-x86_64-vmware-vcenter: Tests completed in 23.08s
12:30:47     win-2008-x86_64-vmware-vcenter: Tests Passed: 3, Failed: 0, Skipped: 0, Pending: 0, Inconclusive: 0
```

**Comment by glennsarti** [comment]
Good enough for me...

hulk

**Pull request merged**